Optimal Steps Model (#26)
PedestrianDynamics/PedestrianDynamics.github.io: content/models/optimal_steps_model/index.md

Reviewer: You should add a Link to this file on the Model-Overview Page by adding

`{{< card link="optimal_steps_model" title="Optimal Steps Model" icon="plus-circle" >}}`

to the the `content/models/_index.md` file.

Author: Done!

---
title: Optimal Steps Model
math: true
---

## Introduction

> The basic idea of the Optimal Steps Model (OSM) is that virtual pedestrians (agents) try to improve their situation with every step. The utility of each position in space is coded in a scalar function called floor field. Utility increases when approaching targets and decreases when getting too close to obstacles and other virtual pedestrians. Physically spoken, agents, when moving, are attracted by targets and repulsed by obstacles and other agents. That is, the negative utility can be interpreted as a potential that agents seek to minimize. The utility, or potential, depend on the geodesic distance to the target and on the proximity to other agents.

Reviewer: In the Sentence:
"Physically spoken, agents, when moving, are attracted by targets and repulsed by obstacles and other agents."

"Physically spoken" doesn't sound right, I would change it to "Physically speaking".

In the Sentence:
"The utility, or potential, depend on the geodesic distance to the target and on the proximity to other agents."
"depend" should be replaced with "depends"

```suggestion
The basic idea of the Optimal Steps Model (OSM) is that virtual pedestrians (agents) try to improve their situation with every step. The utility of each position in space is coded in a scalar function called floor field. Utility increases when approaching targets and decreases when getting too close to obstacles and other virtual pedestrians. Physically speaking, agents, when moving, are attracted by targets and repulsed by obstacles and other agents. That is, the negative utility can be interpreted as a potential that agents seek to minimize. The utility, or potential, depends on the geodesic distance to the target and on the proximity to other agents.
```

Author: Done!

> {{< figure src="figure1.png" caption="Schamtic solution of routing with OSM." >}}

Reviewer: This figure is taken from  "The Heat Method for geodesic distance computation
on 2D domains" however i do not see this mentioned anywhere. Please make sure to add the reference to the source.

All the other figures are copied from other papers too. I am unsure how these figures have to be referenced to be in order with copyright issues but you should be certain before publishing it.

> Agents move by stepping on the position on a circle (or disk) around their current location that optimizes this utility. The circle radius represents each agent's personal maximum stride length, which in turn is linearly correlated to the agent's free-flow speed [[1]](#Seitz2012), that is, an assumed desired speed when the path is free. Thus, agents step towards targets while skirting obstacles and avoiding collisions. 

Reviewer: In the Sentence:
"Agents move by stepping on the position on a circle (or disk) around their current location that optimizes this utility." 
There should be a comma between location & that.
"around their current location, that optimizes this utility."

In the Sentence:
"Thus, agents step towards targets while skirting obstacles and avoiding collisions."
"towards" is British English. In American English it should be "toward".
"Thus, agents step toward targets while skirting obstacles and avoiding collisions."

```suggestion
Agents move by stepping on the position on a circle (or disk) around their current location, that optimizes this utility. The circle radius represents each agent's personal maximum stride length, which in turn is linearly correlated to the agent's free-flow speed [[1]](#Seitz2012), that is, an assumed desired speed when the path is free. Thus, agents step toward targets while skirting obstacles and avoiding collisions. 
```

Author: Done!

> In the remainder of the text, and in the figures, we use the physical interpretation in alignment with the names of parameters in Vadere's implementation of the OSM.

Reviewer: Remove the comma between "text" & "and"
```suggestion
In the remainder of the text and in the figures, we use the physical interpretation in alignment with the names of parameters in Vadere's implementation of the OSM.
```

Author: Done

If we consider the floor field to be static, the value of the potential function at position $x$ equals the negative geodesic distance between $x$ and the target.

The obstacle potential $P_{o,j}(x)$ is defined as 

<!-- $$ -->
<!-- \begin{eqnarray} -->
<!-- P_{o,j}(x) = \left\{ -->
<!-- \begin{array}{ll} -->
<!-- \delta_1(x) + \delta_2(x) & d_{o,j}(x) < r_l \\\ -->
<!-- \delta_1(x)  & r_l \leq d_{o,j}(x) < w_o \\\ -->
<!-- 0 & \, \textrm{otherwise,} \\\ -->
<!-- \end{array} -->
<!-- \right. -->
<!-- \end{eqnarray} -->
<!-- $$ -->

$$
P_{o,j}(x) =
\begin{array}{ll}
\delta_1(x) + \delta_2(x) & d_{o,j}(x) < r_l \\\
\delta_1(x)  & r_l \leq d_{o,j}(x) < w_o \\\
0 &  \textrm{otherwise,} \\\
\end{array}
$$

> where $d_{o,j}(x)$ is the distance to the closest point of the obstacle from position $x$, $w_o$ defined the width of the obstacle repulsion, $r_l$ is the torso radius of agent $l$. The functions $\delta_i$ are defined as

Reviewer: When explaining $d_{l,i}​(x)$ in Line 71 it was explicitly mentioned that the Euclidean distance is used. For better understanding this could be added here too.

Author: Done

where $d_{l,i}(x)$ is the Euclidean distance between agent $l$  and agent $i$. $r_i, r_l$ are the respective torso radii of agent $i,l$. $w_{int}$ and  $w$ are the intimate and the personal space width according to Hall. The functions $\phi_i$ are defined as
$$
\begin{align}
\phi_1({d_{i,j}}) &= \mu \exp{\left[ 4 \left( \left(  \frac{d_{i,j}} {w +r_i +r_l} \right)^{2} - 1\right)^{-1} \right]} \\\
\phi_2({d_{i,j}}) &= \frac{ \mu } {a_p} \exp{\left[ 4 \left( \left(  \frac{d_{i,j}} {w_{int} +r_i +r_l} \right)^{2} - 1\right)^{-1} \right]} \\\
\phi_3({d_{i,j}}) &= 10^3 \exp{\left[ \left( \left(  \frac{d_{i,j}} {r_i +r_l} \right)^{2} - 1\right)^{-1} \right]}
\end{align}
$$

The potential function is based on Hall's theory of interpersonal distances which describes four distance zones around a person [[4]](#Hall). Accordingly, the potential function is defined piece-wise on rings around each agent: a circular core for collision avoidance, a first ring that represents the intimate space, and a second ring that represents personal space. Agents outside the personal zone do not affect other agents' path choice. This is mathematically modeled by setting the  the potential function to zero.

> {{< figure src="figure3.png" caption="Agent's potential versus distance." >}}

Reviewer: This figure might need more Explanation. There are two functions showing seemingly the same thing. What is the difference?

> The value of the potential function in the personal space ring is very low. Thus, this area will be kept free only if agents have ample space to avoid each other [[3]](#Sivers2016). As soon as the space becomes more constricted agents will get closer. This is typical for normal human behavior. In the intimate space ring, the potential function value increases significantly. In crowds, this area is only kept free if the density is low [[3]](#Sivers2016). Finally, to prevent agents from overlapping, the potential is set to a high value (compared to the values for the personal and intimate spaces) in the collision area. The exact definition of the potential function and default parameters implemented in \textit{Vadere} can be found in [[2]](#Kleinmeier2019).

Reviewer: You should somewhere in the mathematical description put a reference to [5] (#Mayr2021), as it was copied almost identically.

> The value of the potential function in the personal space ring is very low. Thus, this area will be kept free only if agents have ample space to avoid each other [[3]](#Sivers2016). As soon as the space becomes more constricted agents will get closer. This is typical for normal human behavior. In the intimate space ring, the potential function value increases significantly. In crowds, this area is only kept free if the density is low [[3]](#Sivers2016). Finally, to prevent agents from overlapping, the potential is set to a high value (compared to the values for the personal and intimate spaces) in the collision area. The exact definition of the potential function and default parameters implemented in \textit{Vadere} can be found in [[2]](#Kleinmeier2019).

Reviewer: In the last Sentence:

\textit is not rendered properly as this is not a latex environment but a markdown one. put '*' before and after for cursive text. However no where else Vedere is put in cursive. I would suggest to either remove the cursive text here or put every Vadere in cursive.

```suggestion
The value of the potential function in the personal space ring is very low. Thus, this area will be kept free only if agents have ample space to avoid each other [[3]](#Sivers2016). As soon as the space becomes more constricted agents will get closer. This is typical for normal human behavior. In the intimate space ring, the potential function value increases significantly. In crowds, this area is only kept free if the density is low [[3]](#Sivers2016). Finally, to prevent agents from overlapping, the potential is set to a high value (compared to the values for the personal and intimate spaces) in the collision area. The exact definition of the potential function and default parameters implemented in Vadere can be found in [[2]](#Kleinmeier2019).
```

Author: Done

- `preLoop()` : set last time step.
- `postLoop()` : shut down update scheme.
- `update()` : set new time step and solve optimization problem to find the next position.

### Usage

Vadere is implemented in Java programming language. Thus, it is available for GNU-/Linux, MacOS and Microsoft Windows. Vadere reads in simulation parameters, like the topography or an agent’s radius, from a human-readable JSON-based text file instead of using a binary format. Also, the simulation results — usually x and y coordinates for each pedestrian and a time step — are written to text files. In this way, users can use text editors to create input files for Vadere and they can open result files from Vadere with 3rd-party software like MATLAB. Furthermore, text files allow users to modify parameters quickly in an automated way. This is essential for studies where parameters must be varied and, thus, thousands of simulations must be run. Performing simulations with Vadere requires three steps. These steps are supported by the graphical user interface. 

{{< figure src="figure5.png" caption="Vadere interface." >}}

### Challenges and limitations

> Using the OSM involves solving many optimization problems. For each step of each agent, a non-trivial optimization problem has to be solved. This optimization is computationally expensive and requires most of the overall computation time. Introducing more complex potential functions complicates the evaluation of the potential function which contributes directly to the computation time of the optimization. Evaluating the potential function at specific points only instead of using an optimization technique such as Nelder Mead can accelate the simulation. Note that when you cut down the number of evaluation points to 8, this will mimic a cellular automaton.

Reviewer: change "accelate" to "accelerate"
```suggestion
Using the OSM involves solving many optimization problems. For each step of each agent, a non-trivial optimization problem has to be solved. This optimization is computationally expensive and requires most of the overall computation time. Introducing more complex potential functions complicates the evaluation of the potential function which contributes directly to the computation time of the optimization. Evaluating the potential function at specific points only instead of using an optimization technique such as Nelder Mead can accelerate the simulation. Note that when you cut down the number of evaluation points to 8, this will mimic a cellular automaton.
```

Author: Done

{{< figure src="figure6.png" caption="Grid discrimination." >}}

> This is aggravated by the fact that a strict event-driven update hinders parallelization of the computation. Therefore, simulating thousands of agents in real-time using the OSM with Vadere is not yet possible. Benedikt Zönnchen worked on the acceleration on his dissertation thesis. 

Reviewer: In the Sentence "Benedikt Zönnchen worked on the acceleration on his dissertation thesis."

change "on his dissertation thesis" to "in his dissertation thesis"

```suggestion
This is aggravated by the fact that a strict event-driven update hinders parallelization of the computation. Therefore, simulating thousands of agents in real-time using the OSM with Vadere is not yet possible. Benedikt Zönnchen worked on the acceleration in his dissertation thesis. 
```

Author: Done
